manager/manager.go

```diff
@@ -388,26 +388,34 @@ func (m *Manager) Run(parent context.Context) error {
 
 	close(m.started)
 
+	var runErr error
 	go func() {
-		err := m.raftNode.Run(ctx)
-		if err != nil {
-			log.G(ctx).WithError(err).Error("raft node stopped")
+		runErr = m.raftNode.Run(ctx)
+		if runErr != nil {
+			log.G(ctx).WithError(runErr).Error("raft node stopped")
 			m.Stop(ctx)
 		}
 	}()
 
-	if err := raft.WaitForLeader(ctx, m.raftNode); err != nil {
+	returnErr := func(err error) error {
+		if err == context.Canceled && runErr != nil {
+			return runErr
```

Comment:
I'm surprised the race detector doesn't complain about this. I think `runErr` should be returned over a channel.